Mesh_3/include/CGAL/Labeled_mesh_domain_3.h

```diff
@@ -310,24 +289,24 @@ class Labeled_mesh_domain_3 :
 
   typedef typename BGT::FT FT;
   typedef BGT Geom_traits;
+  using Impl_details::construct_pair_functor;
 
+  template<typename Func, typename BoundingBox, typename CGAL_NP_TEMPLATE_PARAMETERS>
+  Labeled_mesh_domain_3(const Func& f, const BoundingBox& bbox, const CGAL_NP_CLASS& np = parameters::default_values())
+  :Impl_details(f,bbox,
+                parameters::choose_parameter(parameters::get_parameter(np, internal_np::error_bound), FT(1e-3)),
+                parameters::choose_parameter(parameters::get_parameter(np, internal_np::surface_patch_index), construct_pair_functor()),
+                parameters::choose_parameter(parameters::get_parameter(np, internal_np::null_subdomain_index_param), Null_subdomain_index()),
+                parameters::choose_parameter(parameters::get_parameter(np, internal_np::rng), nullptr))
+  {
+  }
+#ifndef CGAL_NO_DEPRECATED_CODE
+        template<typename Func, typename BoundingBox, typename ... NP_PACK>
+        Labeled_mesh_domain_3(const Func& f, const BoundingBox& bbox, const NP_PACK& ...nps):Labeled_mesh_domain_3(f, bbox, internal_np::combine_named_parameters(nps...))
+        {
+        }
+#endif //CGAL_NO_DEPRECATED_CODE
 
-  BOOST_PARAMETER_CONSTRUCTOR(Labeled_mesh_domain_3,
-                              (Impl_details),
-                              parameters::tag,
-                              (required
```

Comment:
One thing I did not know is that some parameters are mandatory. We should document that. We should also add some `static_assert` making it obvious to the user that a required parameter is missing. See for example `is_default_parameter`.